### PR TITLE
geoserver: fix startup script and add nixos test

### PR DESCRIPTION
**Pull request by imincik**
## Description of changes

* Make sure startup script is not failing even if GEOSERVER_DATA_DIR variable is not set by user
* Add nixos test
* Update maintainer to geospatial team

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by imincik** [comment]
@ofborg build geoserver

**Comment by imincik** [comment]
@ofborg build geoserver

File: pkgs/servers/geospatial/geoserver/default.nix [hunk 1]
```diff
@@ -27,12 +40,16 @@ stdenv.mkDerivation rec {
...
+    tests.geoserver = nixosTests.geoserver;
+  };
+
   meta = with lib; {
     description = "Open source server for sharing geospatial data";
     homepage = "https://geoserver.org/";
     sourceProvenance = with sourceTypes; [ binaryBytecode ];
     license = licenses.gpl2Plus;
-    maintainers = with maintainers; [ sikmir ];
+    maintainers = with maintainers; teams.geospatial.members;
```
Comment (sikmir) [inline pkgs/servers/geospatial/geoserver/default.nix:52]:
```suggestion
    maintainers = teams.geospatial.members;
```

Comment (imincik) [reply pkgs/servers/geospatial/geoserver/default.nix]: Done. Thank you for feedback.

File: nixos/tests/all-tests.nix [hunk 2]
```diff
@@ -320,6 +320,7 @@ in {
   mimir = handleTest ./mimir.nix {};
   garage = handleTest ./garage {};
   gemstash = handleTest ./gemstash.nix {};
+  geoserver = runTest ./geoserver.nix;
```
Comment (imincik) [inline nixos/tests/all-tests.nix:323]: @sikmir, I am not sure if I should use handleTest or runTest here. Do you have some preference ?

**Comment by imincik** [comment]
@sikmir , can we merge this now ?

**Pull request merged**